complex objects/cargo/list/src/categories.rs

```diff
@@ -15,40 +15,36 @@ use std::os::raw::{
 };
 use std::sync::{
     Arc,
-    Mutex
 };
 
-use rusqlite::Connection;
-use time::Timespec;
-
-use items::Item;
-use utils::{
+use ffi_utils::strings::{
     c_char_to_string,
-    read_connection,
     string_to_c_char
 };
+use items::Item;
+use store::Store;
 
 #[derive(Debug)]
 #[repr(C)]
 pub struct CategoryManager {
-    conn: Arc<Mutex<Connection>>,
-    uri: String
+    store: Arc<Store>,
 }
 
 impl CategoryManager {
-    pub fn new(uri: String, conn: Arc<Mutex<Connection>>) -> CategoryManager {
-        CategoryManager {
-            conn: conn,
-            uri: uri
-        }
+    pub fn new(store: Arc<Store>) -> CategoryManager {
```

Comment:
This commit is a bit more than a refactor, no? Might be worth flattening and/or re-splitting.